### PR TITLE
OS related code cleanup

### PR DESCRIPTION
**Pull request by xu-cheng**
cc @bfontaine @mikemcquaid @woodruffw

### CODE REVIEW

File: Library/Homebrew/os/mac/mach.rb [hunk 1]
```diff
@@ -202,3 +199,7 @@ def dylib_id
     mach_metadata.dylib_id
   end
 end
+
+class Pathname
+  include(MachO)
+end
```
Comment (mikemcquaid) [inline Library/Homebrew/os/mac/mach.rb]: I think it's better this lives in `Library/Homebrew/extend/pathname.rb` or `Library/Homebrew/extend/pathname` somewhere.

Comment (xu-cheng) [inline Library/Homebrew/os/mac/mach.rb]: Actually, I move this from `Library/Homebrew/extend/pathname.rb`. What I'm thinking is to put OS related code inside `os` folder.

Comment (mikemcquaid) [inline Library/Homebrew/os/mac/mach.rb]: Yeh, I prefer all monkeypatching happens in `Library/Homebrew/extend/` somewhere.

Comment (xu-cheng) [inline Library/Homebrew/os/mac/mach.rb]: How about `Library/Homebrew/os/mac/extend/` or something?

If we still like to put it in original place, may be as `include(MachO) if OS.mac?`?

If you don't like either of these, I'm happy to keep original form. (Just make sure on the same page for style issue)

Comment (mikemcquaid) [inline Library/Homebrew/os/mac/mach.rb]: How about `Library/Homebrew/extend/os/mac/pathname`?

Comment (xu-cheng) [inline Library/Homebrew/os/mac/mach.rb]: :+1:  will update PR shortly.

Comment (xu-cheng) [inline Library/Homebrew/os/mac/mach.rb]: Updated.

File: Library/Homebrew/os/mac/mach.rb [hunk 2]
```diff
@@ -164,11 +164,10 @@ def initialize(path)
     end
 
     def parse_otool_L_output
-      ENV["HOMEBREW_MACH_O_FILE"] = path.expand_path.to_s
```
Comment (mikemcquaid) [inline Library/Homebrew/os/mac/mach.rb]: I'm hesitant to remove this environment variable unless you're 100% sure there's no reason it was being used before.

**Comment by mikemcquaid** [comment]
One comment otherwise :+1:

File: Library/Homebrew/os/mac/mach.rb (hunk 2 above)
Comment (xu-cheng) [inline Library/Homebrew/os/mac/mach.rb]: I'm 100% sure this is used to escape path. See https://github.com/Homebrew/homebrew/blob/8f9854f81f8064c1428a05efc8d9754e92e259a7/Library/Homebrew/mach.rb#L153

Comment (mikemcquaid) [inline Library/Homebrew/os/mac/mach.rb]: Ok. :+1:

**Pull request closed without merging**